pkg/services/store/object/http.go

```diff
@@ -0,0 +1,165 @@
+	// For now, require admin for everything
+	reqGrafanaAdmin := middleware.ReqSignedIn //.ReqGrafanaAdmin
+
+	// Every * must parse to a GRN (uid+kind)
+	route.Get("/store/*", reqGrafanaAdmin, routing.Wrap(s.doGetOject))
+	route.Get("/raw/*", reqGrafanaAdmin, routing.Wrap(s.doGetRawOject))
+	route.Post("/store/*", reqGrafanaAdmin, routing.Wrap(s.doWriteObject))
+	route.Delete("/store/*", reqGrafanaAdmin, routing.Wrap(s.doDeleteObject))
+	route.Get("/history/*", reqGrafanaAdmin, routing.Wrap(s.doGetHistory))
+	route.Get("/list/*", reqGrafanaAdmin, routing.Wrap(s.doListFolder)) // Simplified version of search -- path is prefix
+	route.Get("/search", reqGrafanaAdmin, routing.Wrap(s.doSearch))
+}
+
+// GRN? path >> UID + kind???
+func parseRequestParams(req *http.Request) (uid string, kind string, params map[string]string) {
+	params = web.Params(req)
+	path := params["*"]
+	idx := strings.LastIndex(path, ".")
+	if idx > 0 {
+		uid = path[:idx]
+		kind = path[idx:]
+	} else {
+		uid = path
+		kind = "?"
+	}
+	return
+}
+
+func (s *httpObjectStore) doGetOject(c *models.ReqContext) response.Response {
+	uid, kind, params := parseRequestParams(c.Req)
+	rsp, err := s.store.Read(c.Req.Context(), &ReadObjectRequest{
+		UID:         uid,
+		Kind:        kind,
+		Version:     params["version"], // ?version = XYZ
+		WithBody:    true,              // ?? allow false?
+		WithSummary: true,              // ?? allow false?
+	})
+	if err != nil {
+		return response.Error(500, "?", err)
+	}
+	if rsp.Object == nil {
+		rsp.Object = &RawObject{
+			UID:      uid,
+			Kind:     "missing!",
+			Modified: time.Now().UnixMilli(),
+		}
+	}
+	return response.JSON(200, rsp)
+}
+
+func (s *httpObjectStore) doGetRawOject(c *models.ReqContext) response.Response {
+	uid, kind, params := parseRequestParams(c.Req)
+	rsp, err := s.store.Read(c.Req.Context(), &ReadObjectRequest{
+		UID:         uid,
+		Kind:        kind,
+		Version:     params["version"], // ?version = XYZ
+		WithBody:    true,
+		WithSummary: false,
+	})
+	if err != nil {
+		return response.Error(500, "?", err)
+	}
+	if rsp.Object != nil && rsp.Object.Body != nil {
+		return response.CreateNormalResponse(
+			http.Header{
+				"Content-Type": []string{"application/json"}, // TODO, based on kind!!!
+			},
+			rsp.Object.Body,
+			200,
+		)
+	}
+	return response.JSON(400, rsp) // ???
+}
+
+func (s *httpObjectStore) doWriteObject(c *models.ReqContext) response.Response {
+	uid, kind, params := parseRequestParams(c.Req)
+	b, err := ioutil.ReadAll(c.Req.Body)
+	if err != nil {
+		return response.Error(400, "error reading body", err)
+	}
+
+	rsp, err := s.store.Write(c.Req.Context(), &WriteObjectRequest{
+		UID:             uid,
+		Kind:            kind,
+		Body:            b,
+		Comment:         params["comment"],
+		PreviousVersion: params["previous"],
+	})
+	if err != nil {
+		return response.Error(500, "?", err)
+	}
+	return response.JSON(200, rsp)
```

Comment:
Should this be a [201 http code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201)?

Reply:
maybe -- 201 is if you create a new object right?

Perhaps:
201: if we created
200: if we updated?
304: write had same bytes as before?

Reply:
yes that seems reasonable assuming we want our create endpoint to also include the update case (vs. splitting those out into separate api calls) IMO having separate create / update endpoints probably is better from maintainability / testing standpoint (i.e. different logic for checking / verifying create object params vs update)

so maybe `doCreateObject` + `doUpdateObject`?

Reply:
Depends mostly if we want the gRPC to have "Write" vs "Create"+"Update" -- I have always been annoyed by APIs that do not essentially do "Upsert" -- though I know options vary on this issue :)